ADD IPs control interface for robot connection in neuronavigation

MarcioCamposJr opened the pull request:
I changed the preferences page to make it possible to add and remove robot connection IPs to make the neuronavigation routine easier.

rmatsuda commented:
Thanks for the PR @MarcioCamposJr! 
I got the following error after delete the init IP, close it and open it again. I suggest to always load the ROBOT_IPS from constants.py

\invesalius3\invesalius\gui\preferences.py", line 1590, in __init__
    choice_IP.SetSelection(robot_ip_options.index(self.robot.robot_ip))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: '169.254.128.103' is not in list

MarcioCamposJr commented:
Ok, I fixed the bugs and also identified others and made the corrections. In addition, I added a new approach to display the connection status with the robot and a pop-up window to confirm the removal of the IP.

**> [!IMPORTANT]: The change directly impacts the functioning of the robot's navigation in Invesalius, so it is important that the versions of Invesalius and https://github.com/biomaglab/tms-robot-control are up to date. The pull request for the version is available at the link: https://github.com/biomaglab/tms-robot-control/pull/17**

rmatsuda commented:
Thanks for the PR @MarcioCamposJr! Merging..
